### PR TITLE
Added adapter setting to appliance network resource

### PR DESCRIPTION
**Pull request by Krishnan-Priyanshu**
# Description
Added adapter setting to appliance network resource

# GitHub Issues
Added adapter setting to appliance network resource

# ISSUE TYPE

- Docs Pull Request
- Feature Pull Request
- Test Pull Request

##### RESOURCE OR DATASOURCE NAME
ome_appliance_network

##### OUTPUT

```paste below

```
##### ADDITIONAL INFORMATION

```paste below

```
# Checklist:

- [x] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [x] I have verified that new and existing unit tests pass locally with my changes
- [x] I have not allowed coverage numbers to degenerate
- [ ] I have maintained at least 90% code coverage
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have maintained backward compatibility

### CODE REVIEW

**Comment by terraform-providers-svc** [comment]
Can one of the admins verify this patch?

File: docs/resources/appliance_network.md [hunk 1]
```diff
@@ -46,11 +91,14 @@ Resource for managing network_setting on OpenManage Enterprise.
 <a id="nestedatt--adapter_setting"></a>
 ### Nested Schema for `adapter_setting`
 
+Required:
+
+- `interface_name` (String) If there are multiple interfaces, network configuration changes can be applied to a single interface using the `interface name` of the NIC.If this option is not specified, Primary interface is chosen by default.
```
Comment (rounak-adhikary) [inline docs/resources/appliance_network.md:96]: Since this is required attribute, ig the 2nd sentence no longer applies.

**Review by rounak-adhikary — COMMENTED** [review]
All else lgtm

**Review by rounak-adhikary — APPROVED** [review]
LGTM

**Review by Krunal-Thakkar — APPROVED** [review]

**Pull request merged**